### PR TITLE
#99 - generate metadata checksums

### PR DESCRIPTION
**Pull request by olenagerasimova**
For #99 I've added checksums generation for `maven-metadata.xml`, tested it and extended test for `AstoMaven`.

### CODE REVIEW

File: src/test/java/com/artipie/maven/asto/AstoMavenTest.java [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+                new ListOf<Matcher<? super XML>>(
+                    // @checkstyle LineLengthCheck (20 lines)
+                    XhtmlMatchers.hasXPath("/metadata/groupId[text() = 'com.test']"),
+                    XhtmlMatchers.hasXPath("/metadata/artifactId[text() = 'logger']"),
+                    XhtmlMatchers.hasXPath("/metadata/versioning/latest[text() = '1.0']"),
+                    XhtmlMatchers.hasXPath("/metadata/versioning/release[text() = '1.0']"),
+                    XhtmlMatchers.hasXPath("/metadata/versioning/versions/version[text() = '1.0']"),
+                    XhtmlMatchers.hasXPath("/metadata/versioning/versions/version[text() = '0.8']"),
+                    XhtmlMatchers.hasXPath("/metadata/versioning/versions/version[text() = '0.9']"),
+                    XhtmlMatchers.hasXPath("metadata/versioning/versions[count(//version) = 3]"),
```
Comment (g4s8) [inline src/test/java/com/artipie/maven/asto/AstoMavenTest.java:100]: @olenagerasimova it seems you have missed leading slash here

File: src/test/java/com/artipie/maven/asto/AstoMavenTest.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+                    XhtmlMatchers.hasXPath("/metadata/versioning/versions/version[text() = '1.0']"),
+                    XhtmlMatchers.hasXPath("metadata/versioning/versions[count(//version) = 1]"),
+                    XhtmlMatchers.hasXPath("/metadata/versioning/lastUpdated")
+                )
+            )
+        );
+    }
+
+    @Test
+    void addsMetadataChecksums() {
```
Comment (g4s8) [inline src/test/java/com/artipie/maven/asto/AstoMavenTest.java:132]: @olenagerasimova don't we need to verify the content of checksum files too?

Comment (olenagerasimova) [reply src/test/java/com/artipie/maven/asto/AstoMavenTest.java:132]: @g4s8 I think we don't need to as it's the job of `RepositoryChecksums#generate()` which is tested in `RepositoryChecksumsTest`. `AstoMaven` here is the user of `RepositoryChecksums` and I think it's enough to check that files were added here

**Comment by olenagerasimova** [comment]
@g4s8 thanks, corrected and answered, take a look please

**Comment by olenagerasimova** [comment]
@g4s8 ping

**Comment by 0crat** [comment]
Job was finished in 24 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Job audit: performer @g4s8[/z](https://www.0crat.com/u/g4s8)[/z](https://www.0crat.com/u/g4s8) didn't make CR comments

**Pull request merged**